app/models/borrow_request.rb

```diff
@@ -0,0 +1,20 @@
+class BorrowRequest < ApplicationRecord
+    belongs_to :user
+    belongs_to :book
+
+    validates :issue_date , presence: true
+    validates :return_due_date , presence: true
+    validates :borrow_start_date , presence: true
+    validates :return_date , presence: true
+
+    validates :validate_unique_accepted_request
+
+  private
+
+  def validate_unique_accepted_request
+    existing_request = BorrowRequest.find_by(book_id: self.book_id, status: 'accepted' , returned: false)
```

Comment:
you should use .present? here as you don't wanna load the db record, you just wanna know if it exists or not